packages/siwe/lib/client.ts

```diff
@@ -189,165 +171,98 @@ export class SiweMessage {
    */
   async verify(
     params: VerifyParams,
-    opts: VerifyOpts = { suppressExceptions: false }
-  ): Promise<SiweResponse> {
-    return new Promise<SiweResponse>((resolve, reject) => {
-      const fail = result => {
-        if (opts.suppressExceptions) {
-          return resolve(result);
-        } else {
-          return reject(result);
-        }
-      };
-
-      const invalidParams: Array<keyof VerifyParams> = checkInvalidKeys<VerifyParams>(params, VerifyParamsKeys);
-      if(invalidParams.length > 0) {
-        fail({
-          success: false,
-          data: this,
-          error: new Error(`${invalidParams.join(', ')} is/are not valid key(s) for VerifyParams.`),
-        })
-      }
+    opts: VerifyOpts = {}
+  ): Promise<void> {
+    const invalidParams: Array<keyof VerifyParams> = checkInvalidKeys<VerifyParams>(params, VerifyParamsKeys);
+    const invalidOpts: Array<keyof VerifyOpts> = checkInvalidKeys<VerifyOpts>(opts, VerifyOptsKeys);
+    const { signature, domain, nonce, time } = params;
+    
+    if (invalidParams.length > 0) {
+      throw new Error(`${invalidParams.join(', ')} is/are not valid key(s) for VerifyParams.`);
+    }
 
-      const invalidOpts: Array<keyof VerifyOpts> = checkInvalidKeys<VerifyOpts>(opts, VerifyOptsKeys);
-      if(invalidParams.length > 0) {
-        fail({
-          success: false,
-          data: this,
-          error: new Error(`${invalidOpts.join(', ')} is/are not valid key(s) for VerifyOpts.`),
-        })
-      }
+    if (invalidOpts.length > 0) {
+      throw new Error(`${invalidOpts.join(', ')} is/are not valid key(s) for VerifyOpts.`);
+    }
+    
+    /** Domain binding */
+    if (domain !== this.domain) {
+      throw new SiweError(SiweErrorType.DOMAIN_MISMATCH, domain, this.domain);
+    }
 
-      const { signature, domain, nonce, time } = params;
-
-      /** Domain binding */
-      if (domain && domain !== this.domain) {
-        fail({
-          success: false,
-          data: this,
-          error: new SiweError(
-            SiweErrorType.DOMAIN_MISMATCH,
-            domain,
-            this.domain
-          ),
-        });
-      }
+    /** Nonce binding */
+    if (nonce !== this.nonce) {
+      throw new SiweError(SiweErrorType.NONCE_MISMATCH, nonce, this.nonce);
+    }
```

Comment:
This could check for a verification option flag (`skipVerifyNonce` or some better named thing)

```
    if (!skipVerifyNonce && nonce !== this.nonce) {
      throw new SiweError(SiweErrorType.NONCE_MISMATCH, nonce, this.nonce);
    }
```